### DISCUSSION TITLE
모바일에서 Scene 이동이 안됩니다.

### DISCUSSION THREAD

**Discussion by ryusiin** [naverz/zepeto-studio-kor#605] [5. Q&A]
![l_93828163_237_1649660912894](https://user-images.githubusercontent.com/5895396/162683572-de696ae9-6079-4ca1-b440-a2ca2575b8ca.png)

현재 캐릭터를 DontDestroyOnLoad 하지 않고, Scene 이동 후 다시 캐릭터를 생성하고 있는데 이 경우 위와 같이 나옵니다.
>> DontDestroyOnLoad를 써도 안되네요...

Scene 이동 후 다시 캐릭터를 생성하는 방법을 쓰시는 분 있나요?

있다면 관련해서 어떻게 하는 지 공유 부탁드립니다ㅠㅜ

    // :: Scene 로드
    public LoadScene(_scene: number): void {
        // :: 처음 시작할 때
        if (_scene === Enums.GetInstance().Scene.LOBBY &&
            this.iCurrentScene === Enums.GetInstance().Scene.LOBBY) {
            this.InitRuler_Lobby();
            return;
        }

        // :: Instantiate 캐릭터 초기화
        App.GetInstane().GetManager_Spawn().ResetInstantiatePlayers();

        // :: 그 이외
        const async: UnityEngine.AsyncOperation = SceneManagement.SceneManager.LoadSceneAsync(_scene);
        async.add_completed((ele) => {
            if (ele.isDone) {
                // :: 현재 Scene 등록
                this.SetCurrentScene(_scene);

                // :: Initialise Ruler
                if (_scene == Enums.GetInstance().Scene.LOBBY) {
                    this.InitRuler_Lobby();
                } else if (_scene == Enums.GetInstance().Scene.IN_GAME) {
                    this.InitRuler_InGame();
                }
            }
        });
    }

**Comment by Zepeto-TechEvangelist** [comment]
안녕하세요. 단순히 동일한 로그인 캐릭터가 씬에 등장하길 원하신다면
각 씬에 ZEPETOPlayers를 추가하고 캐릭터 로드 스크립트를 추가하시면 가능합니다.

**Reply by ryusiin** [reply]
조금 더 자세히 설명 해주실 수 있나요?
혹시나 예제가 될만한 프로젝트가 있으면 공유 부탁 드립니다:)

![l_93828163_239_1649744619053](https://user-images.githubusercontent.com/5895396/162894208-78f8f57e-7d12-4d5d-857b-c8f8b0d94267.png)

Load 스크립트인지는 모르겠지만, 
아래의 코드와 같이 동일한 로그인 캐릭터로 Create를 하고 있는데 Unity Editor 상에서는 작동을 하는데
실제 모바일에서는 위의 그림처럼 저 로딩에서 멈추는 현상이 일어납니다.

    public InstantiatePlayers_WhenLobby() {
        // :: 각 생성
        this.iJoinedPlayers.forEach((_player: Player, _sessionID: string) => {
            // :: Next : 이미 구현한 캐릭터일 경우
            if (this.iInstantiatePlayers.has(_sessionID))
                return;

            const position: UnityEngine.Vector3 = this.ParseToUnityVector(_player.transform.position);
            const rotation: UnityEngine.Vector3 = this.ParseToUnityVector(_player.transform.rotation);
            const spawnInfo: SpawnInfo = this.MakeSpawnInfo(position, rotation);

            // :: 이 캐릭터가 로컬인지 확인
            const isLocal = App.GetInstane().GetManager_Network().GetRoomSessionID()
                === _player.sessionId;

            // :: 플레이어 제작
            ZepetoPlayers.instance
                .CreatePlayerWithUserId(_sessionID, _player.zepetoUserId,
                    spawnInfo, isLocal);

            // :: Instantiate Player
            this.AddInstantiatePlayer(_sessionID, _player);
        });
    }
    public InstantiatePlayers_WhenInGame() {
        // :: 각 생성
        this.iJoinedPlayers.forEach((_player: Player, _sessionID: string) => {
            // :: Next : 이미 구현한 캐릭터일 경우
            if (this.iInstantiatePlayers.has(_sessionID))
                return;

            const position: UnityEngine.Vector3 = this.ParseToUnityVector(_player.transform.position);
            const rotation: UnityEngine.Vector3 = this.ParseToUnityVector(_player.transform.rotation);
            const spawnInfo: SpawnInfo = this.MakeSpawnInfo(position, rotation);

            // :: 이 캐릭터가 로컬인지 확인
            const isLocal = App.GetInstane().GetManager_Network().GetRoomSessionID()
                === _player.sessionId;

            // :: 플레이어 제작
            ZepetoPlayers.instance
                .CreatePlayerWithUserId(_sessionID, _player.zepetoUserId,
                    spawnInfo, isLocal);

            // :: Instantiate Player
            this.AddInstantiatePlayer(_sessionID, _player);
        });
    }

**Reply by Zepeto-TechEvangelist** [reply]
처음 답변 드릴때는 싱글 플레이 기준이었고, 싱글에서는 모바일 씬 이동시 캐릭터 생성이 잘 됩니다.
올려주신 캐릭터 로드 스크립트는 멀티플레이셔서 멀티플레이에서 확인해보니 모바일 씬 이동이 안되는 현상을 재현했습니다.
관련 원인 확인해 보겠습니다.

**Reply by Zepeto-TechEvangelist** [reply]
사용하신 월드 버전과 제페토 앱 버전을 알려주시면 문제 해결에 도움이 될것 같습니다.

**Reply by ryusiin** [reply]
확인 감사합니다:>

버전은 아래와 같습니다.

<Unity 제페토 월드>
월드 버전 : 1.1.7

<제페토 어플리케이션>
어플리케이션 버전 : 3.11.3
업데이트일 : 2022/04/07

**Reply by Zepeto-TechEvangelist** [reply]
안녕하세요, 

이 부분은 씬 이동을 index로 하려고 할 때, 발생하는 문제로 보입니다.
씬 이동을 SceneName(string) 으로 호출하면 정상작동 할 듯 합니다.


선언부 : import {SceneManager} from "UnityEngine.SceneManagement"; 

예제 코드 : SceneManager.LoadScene("SampleScene");
